src/main/java/seedu/duke/logic/Logic.java

```diff
@@ -172,20 +152,69 @@ public void editEntry(HashMap<String, String> commandArguments) throws FinanceBu
         editEntryCommand.execute(financialList);
     }
 
+    /**
+     * Parses the index from the argument string.
+     */
+    private int parseIndex(String indexStr) throws FinanceBuddyException {
+        try {
+            int index = Integer.parseInt(indexStr);
+            if (index <= 0 || index > financialList.getEntryCount()) {
+                throw new FinanceBuddyException(Commons.ERROR_MESSAGE_INVALID_INDEX);
+            }
+            return index;
+        } catch (NumberFormatException e) {
+            throw new FinanceBuddyException(Commons.ERROR_MESSAGE_INVALID_INDEX);
+        }
+    }
+
+    /**
+     * Parses the amount or returns the default amount if null.
+     */
+    private double parseAmountOrDefault(String amountStr, double defaultAmount) throws FinanceBuddyException {
+        try {
+            return (amountStr != null) ? Double.parseDouble(amountStr) : defaultAmount;
+        } catch (NumberFormatException e) {
+            throw new FinanceBuddyException(Commons.ERROR_MESSAGE_NON_NUMBER_AMOUNT);
+        }
+    }
+
+    /**
+     * Parses the date or returns the default date if null.
+     */
+    private String parseDateOrDefault(String dateStr, LocalDate defaultDate) {
+        return (dateStr != null) ? dateStr : defaultDate.format(DateTimeFormatter.ofPattern("dd/MM/yy"));
+    }
+
+    /**
+     * Updates the balance if the entry is an expense.
+     */
+    private void updateExpenseBalance(Expense entry, double newAmount, String newDate) {
+        double oldAmount = entry.getAmount();
+        LocalDate oldDate = entry.getDate();
+        try {
+            budgetLogic.changeBalanceFromExpense(oldAmount, oldDate);
+            budgetLogic.changeBalanceFromExpenseString(-newAmount, newDate);
+        } catch (FinanceBuddyException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Retrieves or parses the category based on input and entry type.
+     */
     private Enum<?> getCategoryFromInput(HashMap<String, String> commandArguments, FinancialEntry entry) {
-        Enum<?> category;
-        String categoryString = commandArguments.get("/c");
-        if (categoryString != null) {
-            category = parseCategory(categoryString, entry);
+        String categoryStr = commandArguments.get("/c");
+        if (categoryStr != null) {
+            return parseCategory(categoryStr, entry);
         } else if (entry instanceof Income) {
-            category = ((Income) entry).getCategory();
+            return ((Income) entry).getCategory();
         } else {
             assert entry instanceof Expense;
-            category = ((Expense) entry).getCategory();
+            return ((Expense) entry).getCategory();
         }
-        return category;
     }
```

Comment:
Perhaps can just use if statements instead of if-else? Since each condition will return